### PR TITLE
INTGR-290: As a Geotab SDK site user, I would like to see examples in Javascript, C#, & Python for MyGeotab SDK methods [Part 3]

### PR DESCRIPTION
**Pull request by PCGeotab**
https://jira.geotab.com/browse/INTGR-290

Methods to write examples for:

GetTimeZones
Javascript already exists? Yes
GetVersion
Javascript already exists? Yes
GetVersionInformation
Javascript already exists? Yes
OptimizeWaypoints
Javascript already exists? Yes
Remove
Javascript already exists? Yes
Set
Javascript already exists? Yes
UploadMediaFile
Javascript already exists? No
DownloadMediaFile
Javascript already exists? No

### CODE REVIEW

**Comment by kmets89** [comment]
let's hold off on this until Joseph's branch is done so all changes are visible

File: src/pages/myGeotab/apiReference/methodsCodeSamples.ts [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+  },
+  GetVersion: {
+    javascript: "",
+    csharp: `await api.CallAsync<string>("GetVersion");`,
+    python: "api.call('GetVersion')",
+    java: "/* Work in progress - Coming soon */",
+  },
+  GetVersionInformation: {
+    javascript: "",
+    csharp: `await api.CallAsync<object>("GetVersionInformation");`,
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methodsCodeSamples.ts:18]:
```suggestion
    csharp: `await api.CallAsync<VersionInformation>("GetVersionInformation");`,
```

File: src/pages/myGeotab/apiReference/methodsCodeSamples.ts [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+    python: "api.call('GetVersionInformation')",
+    java: "/* Work in progress - Coming soon */",
+  },
+  OptimizeWaypoints: {
+    javascript: "",
+    csharp: `
+    var waypointsList = new List<Waypoint>
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methodsCodeSamples.ts:25]: Tabbing needs to be fixed, missing two waypoints, and missing descriptions for each waypoint.

```suggestion
    IList<Waypoint> waypointsList = new List<Waypoint>
```

File: src/pages/myGeotab/apiReference/methodsCodeSamples.ts [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+    {
+        new Waypoint(new Coordinate(-80.0969, 43.2755), 0),
+        new Waypoint(new Coordinate(-80.0735, 43.298), 1),
+        new Waypoint(new Coordinate(-80.0969, 43.2755), 2)
+    };
+
+    await api.CallAsync<List<Waypoint>>("OptimizeWaypoints", new {waypoints = waypointsList});    
+    `,
+    python: `api.call("OptimizeWaypoints", waypoints=[
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methodsCodeSamples.ts:34]: Fix tabbing

File: src/pages/myGeotab/apiReference/methodsCodeSamples.ts [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+                "sequence": 4,
+                "description": "E"
+            }
+          ])`,
+    java: "/* Work in progress - Coming soon */",
+  },
+  Remove: {
+    javascript: "",
+    csharp: `
+    Console.WriteLine("Enter the name of the zone that you want to remove:");
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methodsCodeSamples.ts:66]: Fix tabbing

File: src/pages/myGeotab/apiReference/methodsCodeSamples.ts [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+    var zones = await api.CallAsync<List<Zone>>("Get", typeof(Zone), new { search = new { name = $"%{zoneName}%"}});
+
+    if (zones.Count > 0) { 
+         await api.CallAsync<Zone>("Remove", typeof(Zone), new { entity = new { id = zones[0].Id } });
+    } else {
+        Console.WriteLine("There is no zone with such name");
+    }
+    `,
+    python: `zone_name = input("Enter the name of the zone that you want to remove: ")
+
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methodsCodeSamples.ts:78]: Fix tabbing

File: src/pages/myGeotab/apiReference/methodsCodeSamples.ts [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+        if zones_response:
+            remove_response = api.remove("Zone",entity = {"id": zones_response[0]["id"]})
+        else:
+            print("There is no zone with such name.")`,
+    java: "/* Work in progress - Coming soon */",
+  },
+  Set: {
+    javascript: "",
+    csharp: `
+    Console.WriteLine("Enter current zone name: ");
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methodsCodeSamples.ts:90]: Fix tabbing

File: src/pages/myGeotab/apiReference/methodsCodeSamples.ts [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+    string currentZoneName = Console.ReadLine();
+
+    Console.WriteLine("Enter new zone name: ");
+    string newZoneName = Console.ReadLine();
+
+    var zones = await api.CallAsync<List<Zone>>("Get", typeof(Zone), new { search = new { name = $"%{currentZoneName}%"}});
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methodsCodeSamples.ts:96]:
```suggestion
    IList<Zone> zones = await api.CallAsync<List<Zone>>("Get", typeof(Zone), new { search = new { name = $"%{currentZoneName}%"}});
```

File: src/pages/myGeotab/apiReference/methodsCodeSamples.ts [hunk 8]
```diff
@@ -0,0 +1,331 @@
...
+
+    if (zones.Count > 0) {
+        var zone = zones[0];
+        zone.Name = newZoneName;
+        await api.CallAsync<Zone>("Set", typeof(Zone), new { entity = zone });
+    }
+    `,
+    python: `current_zone_name = input("Enter current zone name: ")
+        new_zone_name = input("Enter new zone name: ")
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methodsCodeSamples.ts:105]: Fix tabbing

File: src/pages/myGeotab/apiReference/methodsCodeSamples.ts [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+          # Stream the content to a file
+          with open(filename, 'wb') as file:
+              for chunk in response.iter_content(chunk_size=1024 * 1024): 
+                  if chunk:
+                      file.write(chunk)
+    
+          print(f"File downloaded and saved as {filename}.")
+        `,
+    java: "/* Work in progress - Coming soon */",
+  },
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methodsCodeSamples.ts:330]: Fix tabbing

File: src/pages/myGeotab/apiReference/methodsCodeSamples.ts [hunk 10]
```diff
@@ -0,0 +1,331 @@
...
+        Driver = null, // possible to link to a driver
+        Thumbnails = null // media files can serve as thumbnails for other media files
+    };
+
+    mediaFile.Id = await api.CallAsync<Id>("Add", typeof(MediaFile), new { entity = mediaFile});
+
+    // Upload the media file binary
+    Console.WriteLine("Uploading file...");
+
+    string inputfile = ""<your media file here>"; // the path to the file to upload
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methodsCodeSamples.ts:187]:
```suggestion
    string inputfile = "<your media file here>"; // the path to the file to upload
```

File: src/pages/myGeotab/apiReference/methodsCodeSamples.ts [hunk 11]
```diff
@@ -0,0 +1,331 @@
...
+        });        
+        `,
+    csharp: `
+    var solutionId = Id.Create(Guid.NewGuid()); // the unique solution ID of the integration
+
+    Console.WriteLine("Enter the name of the file (must have extension): ");
+    string mediaFileName = Console.ReadLine();
+
+    // -- Add a media file object which describes the binary file
+    var mediaFile = new MediaFile
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methodsCodeSamples.ts:170]:
```suggestion
    MediaFile mediaFile = new MediaFile
```

File: src/pages/myGeotab/apiReference/methodsCodeSamples.ts [hunk 12]
```diff
@@ -0,0 +1,331 @@
...
+    axios.post(api_path, formData)
+        .then(response => {
+            console.log("Success:", response.data);
+        })
+        .catch(error => {
+            console.error("Error:", error.response ? error.response.data : error.message);
+        });        
+        `,
+    csharp: `
+    var solutionId = Id.Create(Guid.NewGuid()); // the unique solution ID of the integration
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methodsCodeSamples.ts:164]:
```suggestion
    Id solutionId = Id.Create(Guid.NewGuid()); // the unique solution ID of the integration
```

File: src/pages/myGeotab/apiReference/methodsCodeSamples.ts [hunk 13]
```diff
@@ -0,0 +1,331 @@
...
+    };
+
+    mediaFile.Id = await api.CallAsync<Id>("Add", typeof(MediaFile), new { entity = mediaFile});
+
+    // Upload the media file binary
+    Console.WriteLine("Uploading file...");
+
+    string inputfile = ""<your media file here>"; // the path to the file to upload
+
+    var file = new FileInfo(inputfile);
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methodsCodeSamples.ts:189]:
```suggestion
    FileInfo file = new FileInfo(inputfile);
```

File: src/pages/myGeotab/apiReference/methodsCodeSamples.ts [hunk 14]
```diff
@@ -0,0 +1,331 @@
...
+        .catch(error => {
+            console.error("Error:", error.response ? error.response.data : error.message);
+        });
+
+    `,
+    csharp: `
+    Console.WriteLine("Enter the id of the media file:");
+    var mediafileId = Id.Create(Console.ReadLine());
+
+    var mediaFile = (await api.CallAsync<IEnumerable<MediaFile>>("Get", typeof(MediaFile), new { search = new MediaFileSearch(mediafileId) })).FirstOrDefault();
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methodsCodeSamples.ts:278]:
```suggestion
    MediaFile mediaFile = (await api.CallAsync<IEnumerable<MediaFile>>("Get", typeof(MediaFile), new { search = new MediaFileSearch(mediafileId) })).FirstOrDefault();
```

File: src/pages/myGeotab/apiReference/methodsCodeSamples.ts [hunk 15]
```diff
@@ -0,0 +1,331 @@
...
+            console.log("Success:", response.data);
+        })
+        .catch(error => {
+            console.error("Error:", error.response ? error.response.data : error.message);
+        });
+
+    `,
+    csharp: `
+    Console.WriteLine("Enter the id of the media file:");
+    var mediafileId = Id.Create(Console.ReadLine());
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methodsCodeSamples.ts:276]:
```suggestion
    Id mediafileId = Id.Create(Console.ReadLine());
```

**Review by jyalenkatian — CHANGES_REQUESTED** [review]

**Review by jyalenkatian — CHANGES_REQUESTED** [review]

**Review by jyalenkatian — APPROVED** [review]

**Pull request merged**